### DISCUSSION TITLE
InvalidFieldRelationship Error When Declaring Relationship to Nested Field in Collection

### DISCUSSION THREAD

**Discussion by faizur11786** [payloadcms/payload#3089] [Feature Requests & Ideas]
I'm encountering an issue where I'm receiving an `InvalidFieldRelationship` error. I'm trying to establish a relationship from one collection to a nested field within another collection.

Here is my Balance collection configuration:

```ts
const Balance: CollectionConfig = {
  slug: 'balances',

  fields: [
    {
      name: 'user',
      type: 'relationship',
      relationTo: ['users', 'admins'],
    },
    {
      name: 'items',
      label: 'NFT Balances',
      type: 'array',
      fields: [
        {
          type: 'row',
          fields: [
            {
              name: 'asset',
              label: 'NFT / Token ID',
              type: 'relationship',
              relationTo: 'nfts',
              required: true,
            },
            {
              name: 'amount',
              label: 'Amount',
              type: 'text',
              required: true,
            },
          ],
        },
      ],
    },
  ],
};
```

I then want to use it in another collection like this:

```ts
{
  name: 'metadata',
  label: 'NFT',
  type: 'relationship',
  relationTo: 'balances.items',
}
```

But this gives me the InvalidFieldRelationship error: "Field NFT has invalid relationship 'balances.items'."

I would appreciate any help or guidance on how to establish a relationship to a nested field within another collection, or if this is an unintended limitation, for this issue to be addressed. Thanks in advance!

**Comment by DanRibbens** [comment]
Hi @faizur11786, 
This isn't going to work: `relationTo: 'balances.items'`. Relationship fields have to use the exact relationship slug in the `relationTo`. You'd have to first relate to balances and then have another field to specify which items. This could be possible but I don't think it is the best way.

It seems to me that you should break your up your balances collection so that instead of an array you have an `assets` collection with relationships to users and nfts along with the amount. Then you can create the metadata relationship field in the other collection with `relationTo: 'assets'`.

Alternatively, could the metadata field be placed in the items array as another field?

Does this help?

**Reply by faizur11786** [reply]
hank you for your insightful response, @DanRibbens.

I now understand the constraints of the `relationTo` field and that it only works directly with collections. I initially assumed it might be feasible to establish a relationship with an array item because each item carries a unique identifier. I imagined this could facilitate a dropdown-like selection for relationships.

To address your follow-up question on the placement of the `metadata` field in the `items` array: my requirements are slightly different. The `metadata` object originates from an entirely separate collection. Hence, direct integration into the `items` array may not serve my purpose optimally.

However, I do appreciate your suggestion about restructuring the `balances` collection. Instead of using an array, having an `assets` collection with relationships to `users` and `nfts`, including the `amount`, sounds like a viable alternative. I will consider this adjustment and see how it fits with my overall design.

If you want I can show both collection.

Thank you for taking the time to provide these clarifications and suggestions!

**Reply by faizur11786** [reply]
`balances` collection fields:
```ts
fields: [
    {
      name: 'user',
      type: 'relationship',
      relationTo: ['users', 'admins'],
      unique: true,
      required: true,
      index: true,
    },
    {
      name: 'balances',
      label: 'NFT Balances',
      type: 'array',
      fields: [
        {
          type: 'row',
          fields: [
            {
              name: 'asset',
              label: 'NFT / Token ID',
              type: 'relationship',
              relationTo: 'sokos-nfts',
              index: true,
              required: true,
            },
            {
              name: 'amount',
              label: 'Amount',
              type: 'text',
              required: true,
            },
            {
              name: 'listedOn',
              label: 'Listed On',
              type: 'relationship',
              index: true,
              relationTo: 'lists',
            },
          ],
        },
      ],
    },
  ],
```

different collection fields:
 ```ts
fields: [
    {
      name: 'nft',
      label: 'My NFT',
      type: 'relationship',
      relationTo: 'balances',
      required: true,
    },
    {
      name: 'listingId',
      label: 'Listing Id',
      type: 'text',
      admin: {
        readOnly: true,
        position: 'sidebar',
      },
    },
    {
      name: 'createdBy',
      label: 'Created By',
      type: 'relationship',
      relationTo: ['admins', 'users'],
      admin: {
        readOnly: true,
        position: 'sidebar',
      },
    },
  ],
```
